device/core/src/client_diagnostic.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { Message, errors } from 'azure-iot-common';
+import { Twin } from './twin';
+import * as dbg from 'debug';
+const debug = dbg('azure-iot-device:Diagnostics');
+
+/**
+ * Data structure to store diagnostic property data
+ */
+export class Diagnostics {
```

Comment:
>Diagnostics [](start = 13, length = 11)

if the class is Diagnostics the file should be named diagnotics.ts (drop the client_ prefix)